pkg/cmd/component/bootstrap_new.go

```diff
@@ -0,0 +1,326 @@
+	if err != nil {
+		return fmt.Errorf("%v Target Port is not valid", io.TargetPort)
+	}
+	if io.PrivateRepoURLDriver != "" {
+		if !supportedDrivers.supported(io.PrivateRepoURLDriver) {
+			return fmt.Errorf("invalid driver type: %q", io.PrivateRepoURLDriver)
+		}
+	}
+	err = setAccessToken(io)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Checking the mandatory flags & reusing missingFlagErr in Bootstrap.go
+func checkMandatoryFlags(flags map[string]string) error {
+	missingFlags := []string{}
+	mandatoryFlags := []string{componentNameFlag, appliactionNameFlag, gitRepoURLFlag, secretFlag}
+	for _, flag := range mandatoryFlags {
+		if flags[flag] == "" {
+			missingFlags = append(missingFlags, fmt.Sprintf("%q", flag))
+		}
+	}
+	if len(missingFlags) > 0 {
+		return missingFlagErr(missingFlags)
+	}
+	return nil
+}
+
+func missingFlagErr(flags []string) error {
+	return fmt.Errorf("required flag(s) %s not set", strings.Join(flags, ", "))
+}
+
+//Interactive mode for Bootstrap-mew Command
+func initiateInteractiveModeForBootstrapNewCommand(io *BootstrapNewParameters, cmd *cobra.Command) error {
+	log.Progressf("\nStarting interactive prompt\n")
+	//Checks for mandatory flags
+	promp := !ui.UseDefaultValuesComponent()
+	if io.ApplicationName == "" {
+		io.ApplicationName = ui.AddApplicationName()
+	}
+	if io.ComponentName == "" {
+		io.ComponentName = ui.AddComponentName()
+	}
+	if io.GitRepoURL == "" {
+		io.GitRepoURL = ui.EnterGitRepoURL()
+	}
+	io.GitRepoURL = utility.AddGitSuffixIfNecessary(io.GitRepoURL)
+	if !isKnownDriverURL(io.GitRepoURL) {
+		io.PrivateRepoURLDriver = ui.SelectPrivateRepoDriver()
+		host, err := accesstoken.HostFromURL(io.GitRepoURL)
+		if err != nil {
+			return fmt.Errorf("failed to parse the gitops url: %w", err)
+		}
+		identifier := factory.NewDriverIdentifier(factory.Mapping(host, io.PrivateRepoURLDriver))
+		factory.DefaultIdentifier = identifier
+	}
+	// We are checking if any existing token is present.
+	//If not we ask the uer to pass the token.
+	//EnterGitSecret is just validating length for now.
+	secret, err := accesstoken.GetAccessToken(io.GitRepoURL)
+	if err != nil && err != keyring.ErrNotFound {
+		return err
+	}
+	if secret == "" { // We must prompt for the token
+		if io.Secret == "" {
+			io.Secret = ui.EnterGitSecret(io.GitRepoURL)
+		}
+		if !cmd.Flag("save-token-keyring").Changed {
+			io.SaveTokenKeyRing = ui.UseKeyringRingSvc()
+		}
+		setAccessToken(io)
+	} else {
+		io.Secret = secret
+	}
+
+	//Optional flags
+	if promp {
+		io.TargetPort = ui.AddTargetPort()
+	}
+
+	if !cmd.Flag("push-to-git").Changed && promp {
+		io.PushToGit = ui.SelectOptionPushToGit()
+	}
+
+	outputPathOverridden := cmd.Flag("output").Changed
+	if !outputPathOverridden {
+		// Override the default path to be ./{gitops repo name}
+		repoName, err := repoFromURL(io.GitRepoURL)
+		if err != nil {
+			repoName = "gitops"
+		}
+		io.Output = filepath.Join(".", repoName)
```

Comment:
I think we should change the default to `./` because the common library will create the folder (with app name as the folder) and inside that folder will contain the generated files.  In that folder is what gets pushed to git.  So we should avoid having two levels of folders and so the repoName folder is unnecessary.  We will have to check if the overwrite behavior is affected by this change, since we will be checking for the existence of the appname folder instead.